crates/typst/src/export/render.rs

```diff
@@ -362,15 +362,21 @@ fn render_outline_glyph(
         let mh = bitmap.height;
 
         let Paint::Solid(color) = text.fill;
-        let c = color.to_rgba();
+        let [r, g, b, _] = color.to_rgba().to_vec4();
 
         // Pad the pixmap with 1 pixel in each dimension so that we do
         // not get any problem with floating point errors along their border
         let mut pixmap = sk::Pixmap::new(mw + 2, mh + 2)?;
         for x in 0..mw {
             for y in 0..mh {
                 let alpha = bitmap.coverage[(y * mw + x) as usize];
-                let color = sk::ColorU8::from_rgba(c.r, c.g, c.b, alpha).premultiply();
+                let color = sk::ColorU8::from_rgba(
+                    (r * 255.0).round() as u8,
+                    (g * 255.0).round() as u8,
+                    (b * 255.0).round() as u8,
+                    alpha,
+                )
+                .premultiply();
```

Comment:
This is on a _very_ hot path. Perhaps the compiler is smart enough to make the float calculations only once, but I wouldn't bet on it. Could you extract them?